### PR TITLE
[Enhancement] Compatible with  GlueCatalog  table that without StorageDescriptor for iceberg table

### PR DESCRIPTION
**Pull request by zombee0**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 10 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hive/glue/util/HiveTableValidator.java|0|7|00.00%|[43, 44, 45, 46, 47, 49, 50]|
|:large_blue_circle:|com/starrocks/connector/hive/glue/converters/CatalogToHiveConverter.java|0|3|00.00%|[160, 161, 162]|

File: fe/fe-core/src/main/java/com/starrocks/connector/hive/glue/converters/CatalogToHiveConverter.java [hunk 1]
```diff
@@ -152,7 +154,13 @@ public static Table convertTable(com.amazonaws.services.glue.model.Table catalog
         Date lastAccessedTime = catalogTable.getLastAccessTime();
         hiveTable.setLastAccessTime(lastAccessedTime == null ? 0 : (int) (lastAccessedTime.getTime() / 1000));
         hiveTable.setRetention(catalogTable.getRetention());
-        hiveTable.setSd(convertStorageDescriptor(catalogTable.getStorageDescriptor()));
+        // for iceberg table, don't need to set StorageDescriptor
+        // just use metadata location in parameters that checked in HiveTableValidator
+        // TODO(zombee0), check hudi deltalake
+        if (catalogTable.getParameters() == null || catalogTable.getParameters().get(TABLE_TYPE_PROP) == null ||
+                (!catalogTable.getParameters().get(TABLE_TYPE_PROP).equalsIgnoreCase(ICEBERG_TABLE_TYPE_VALUE))) {
+            hiveTable.setSd(convertStorageDescriptor(catalogTable.getStorageDescriptor()));
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/connector/hive/glue/converters/CatalogToHiveConverter.java:162]: For iceberg table, is `catalogTable.getStorageDescriptor()` is null ?

Comment (zombee0) [reply fe/fe-core/src/main/java/com/starrocks/connector/hive/glue/converters/CatalogToHiveConverter.java:162]: Yes, some case such as dealed by trino

File: fe/fe-core/src/main/java/com/starrocks/connector/hive/glue/converters/CatalogToHiveConverter.java [hunk 2]
```diff
@@ -152,7 +154,13 @@ public static Table convertTable(com.amazonaws.services.glue.model.Table catalog
         Date lastAccessedTime = catalogTable.getLastAccessTime();
         hiveTable.setLastAccessTime(lastAccessedTime == null ? 0 : (int) (lastAccessedTime.getTime() / 1000));
         hiveTable.setRetention(catalogTable.getRetention());
-        hiveTable.setSd(convertStorageDescriptor(catalogTable.getStorageDescriptor()));
+        // for iceberg table, don't need to set StorageDescriptor
+        // just use metadata location in parameters that checked in HiveTableValidator
+        // TODO(zombee0), check hudi deltalake
+        if (catalogTable.getParameters() == null || catalogTable.getParameters().get(TABLE_TYPE_PROP) == null ||
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/hive/glue/converters/CatalogToHiveConverter.java:160]: It's better to abstract a method named isIcebergTable()

Comment (zombee0) [reply fe/fe-core/src/main/java/com/starrocks/connector/hive/glue/converters/CatalogToHiveConverter.java]: done

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 10 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hive/glue/util/HiveTableValidator.java|0|8|00.00%|[43, 44, 45, 47, 48, 60, 61, 62]|
|:large_blue_circle:|com/starrocks/connector/hive/glue/converters/CatalogToHiveConverter.java|0|2|00.00%|[159, 160]|

**Review by stephen-shelby — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**